components/server/src/user/user-controller.ts

```diff
@@ -575,7 +575,13 @@ export class UserController {
             await this.userService.updateUserIdentity(user, additionalIdentity, additionalToken);
         }
 
-        // const { isBlocked } = tosFlowInfo; // todo@alex: this setting is in conflict with the env var
+        const { isBlocked } = tosFlowInfo;
+        if (isBlocked) {
+            await TosFlow.clear(req.session);
+            await AuthFlow.clear(req.session);
+
+            res.status(403);
```

Comment:
@geropl, this won't be sticky.
the user account was already created in L.567. 
there the called `updateNewUserAfterTos` was the original place to set `User.isBlocked` attribute from `tosFlowInfo.isBlocked`, AFAIR. reinstating this should do the trick.

also, sending response here should be followed by a return, otherwise there will be more attempts to send in the followings.

this is a confusing line of comment which was a left over from the blocking attempts on staging envs.

Reply:
Talked to Alex , the result is pushed as condensed [commit](https://github.com/gitpod-io/gitpod/pull/9385/commits/2a113294d3cace825d19abfeb5d2a5644c381285).